src/zjs_util.c

```diff
@@ -180,6 +180,42 @@ bool zjs_obj_get_int32(jerry_value_t obj, const char *name, int32_t *num)
     return true;
 }
 
+void zjs_copy_jstring(jerry_value_t jstr, char *buffer, jerry_size_t *maxlen)
+{
+    jerry_size_t len = 0;
+    jerry_size_t size = jerry_get_string_size(jstr);
+    if (maxlen && size <= *maxlen) {
+        len = jerry_string_to_char_buffer(jstr, (jerry_char_t *)buffer, size);
+    }
+    buffer[len] = '\0';
+}
+
+char *zjs_alloc_from_jstring(jerry_value_t jstr, jerry_size_t *maxlen)
+{
+    jerry_size_t size = jerry_get_string_size(jstr);
+    char *buffer = zjs_malloc(size + 1);
```

Comment:
No, what this function will do is allocate the full string but then truncate it to the maxlen and still return successfully. I wrote it to replace code in grove_lcd.c and that's what that code was doing.